API 17 - crash on launch from develop branch

# Issue
When trying to launch the application on device with API 17 application crashes due to the below RuntimeException:

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.automattic.simplenote.debug/com.automattic.simplenote.NotesActivity}: java.lang.IllegalStateException: This app has been built with an incorrect configuration. Please configure your build for VectorDrawableCompat.
 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2186)
 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2236)
 at android.app.ActivityThread.access$600(ActivityThread.java:145)
 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1238)
 at android.os.Handler.dispatchMessage(Handler.java:99)
 at android.os.Looper.loop(Looper.java:137)
 at android.app.ActivityThread.main(ActivityThread.java:5099)
 at java.lang.reflect.Method.invokeNative(Native Method)
 at java.lang.reflect.Method.invoke(Method.java:511)
 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:803)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:570)
 at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132)
 at dalvik.system.NativeStart.main(Native Method)
 Caused by: java.lang.IllegalStateException: This app has been built with an incorrect configuration. Please configure your build for VectorDrawableCompat.
 at android.support.v7.widget.AppCompatDrawableManager.checkVectorDrawableSetup(AppCompatDrawableManager.java:692)
 at android.support.v7.widget.AppCompatDrawableManager.getDrawable(AppCompatDrawableManager.java:186)
 at android.support.v7.widget.TintTypedArray.getDrawableIfKnown(TintTypedArray.java:77)
 at android.support.v7.app.AppCompatDelegateImplBase.<init>(AppCompatDelegateImplBase.java:83)
 at android.support.v7.app.AppCompatDelegateImplV7.<init>(AppCompatDelegateImplV7.java:146)
 at android.support.v7.app.AppCompatDelegateImplV11.<init>(AppCompatDelegateImplV11.java:28)
 at android.support.v7.app.AppCompatDelegateImplV14.<init>(AppCompatDelegateImplV14.java:41)
 at android.support.v7.app.AppCompatDelegate.create(AppCompatDelegate.java:193)
 at android.support.v7.app.AppCompatDelegate.create(AppCompatDelegate.java:173)
 at android.support.v7.app.AppCompatActivity.getDelegate(AppCompatActivity.java:511)
 at android.support.v7.app.AppCompatActivity.onCreate(AppCompatActivity.java:71)
 at com.automattic.simplenote.NotesActivity.onCreate(NotesActivity.java:106)
 at android.app.Activity.performCreate(Activity.java:5117)
 at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1081)
 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2150)
 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2236) 
 at android.app.ActivityThread.access$600(ActivityThread.java:145) 
 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1238) 
 at android.os.Handler.dispatchMessage(Handler.java:99) 
 at android.os.Looper.loop(Looper.java:137) 
 at android.app.ActivityThread.main(ActivityThread.java:5099) 
 at java.lang.reflect.Method.invokeNative(Native Method) 
 at java.lang.reflect.Method.invoke(Method.java:511) 
 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:803) 
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:570) 
 at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132) 
 at dalvik.system.NativeStart.main(Native Method)
```

# Solution
Based on [this](http://stackoverflow.com/a/38150219) StackOverflow answer simply updating to latest Gradle plugin version should be enough to fix the issue. I checked this and it indeed solves the issue.

--- a/Simplenote/build.gradle
+++ b/Simplenote/build.gradle
@@ -4,7 +4,7 @@ buildscript {
         maven { url 'https://maven.fabric.io/repo' }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.5.0'
+        classpath 'com.android.tools.build:gradle:2.2.0'
     }
 }
 
@@ -21,11 +21,11 @@ repositories {
 
 dependencies {
     compile "com.simperium.android:simperium:0.6.8"
-    compile "com.android.support:appcompat-v7:23.3.0"
-    compile 'com.android.support:design:23.3.0'
-    compile 'com.android.support:support-vector-drawable:23.3.0'
-    compile "com.google.android.gms:play-services-analytics:8.3.0"
-    compile "com.google.android.gms:play-services-wearable:8.3.0"
+    compile "com.android.support:appcompat-v7:24.2.1"
+    compile 'com.android.support:design:24.2.1'
+    compile 'com.android.support:support-vector-drawable:24.2.1'
+    compile "com.google.android.gms:play-services-analytics:9.6.1"
+    compile "com.google.android.gms:play-services-wearable:9.6.1"
     compile 'org.wordpress:passcodelock:1.3.0'
     compile 'com.automattic:tracks:1.1.0'
     compile 'com.loopj.android:android-async-http:1.4.9'
@@ -38,15 +38,15 @@ version "1.5.1"
 
 android {
 
-    compileSdkVersion 23
-    buildToolsVersion "23.0.3"
+    compileSdkVersion 24
+    buildToolsVersion "24.0.3"
 
     defaultConfig {
         applicationId "com.automattic.simplenote"
         versionName project.version
         versionCode 53
         minSdkVersion 15
-        targetSdkVersion 23
+        targetSdkVersion 24
 
         buildConfigField "String", "SIMPERIUM_APP_ID", "\"${project.simperiumAppId}\""
         buildConfigField "String", "SIMPERIUM_APP_KEY", "\"${project.simperiumAppKey}\""

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
-#Tue Dec 02 12:00:11 PST 2014
+#Tue Oct 04 19:22:26 CEST 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip

